pkg/localstorev2/subscribe_push.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storer
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/localstorev2/internal/upload"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const subscribePushEventKey = "subscribe-push"
+
+func (db *DB) SubscribePush(ctx context.Context) (chunks chan swarm.Chunk, stop func()) {
+	chunks = make(chan swarm.Chunk)
+
+	trigger, unsub := db.events.Subscribe(subscribePushEventKey)
+	defer unsub()
```

Comment:
the unsub should be at the start of the goroutine above

Reply:
moved